steamtinkerlaunch

```diff
@@ -18243,6 +18249,11 @@ function checkIntDeps {
 		if [ -z "$YAD" ]; then
 			YAD="$(command -v "yad")"
 		fi
+		
+		if [ ! "$YAD" ]; then
```

Comment:
Instead of doing this check, we can probably just improve the missing Yad logging in `checkIntDeps`.

Reply:
The problem as you saw if yad is not found I get also some notification and fails the version detection.
So maybe we can improve to just exclude the CLI case when a parameter is used.

Reply:
> So maybe we can improve to just exclude the CLI case when a parameter is used.

I don't know if this will work as SteamTinkerLaunch will call itself from the UI too, for example it might called `mo2 start` from the UI.

I think some loggings and (I assume harmless?) errors are fine. Although Yad really *should* be installed to use SteamTinkerLaunch fully, since a lot of it still uses Yad. For example, the `ansg` may use Yad if called with some or none of the required parameters, but will not if all the parameters are supplied.

I don't know that quitting if Yad is not installed is necessary though, same with the other dependencies (except Steam I guess). We can just let STL fall over if dependencies are missing.

Reply:
I think for now, Yad being mandatory is the path of least resistence. I don't know if the UI will ever be reorganised in the way to avoid Yad being a dependency, since STL calling itself is by design, but for sure this is something to look at in the future.

Reply:
Well it is an hard dependency so we decided, you can just approve the PR if there isn't anything else to do.

Reply:
I think we can remove the Yad check in this way. We already have logging for a missing Yad and I don't think we need to exit on Yad being missing :-)

Reply:
In my opinion we should exit if not found other wise you can get errors in console or notifications that makes no sense.
In this way the tool doesn't keep following do the rest of the program.

Reply:
Reading through `checkIntDeps` agaiin, I think your check is still unnecessary.
1. If Yad is missing, we catch it with `DEPSMISSING=1` in the `if [ ! -x "$(command -v "$YAD")" ]; then` block.
2. At the bottom of `checkIntDeps`, we will exit if any dependencies are missing using `if [ "$DEPSMISSING" -eq 1 ]; then`.
3. We will continue to correctly skip this check if `SKIPINTDEPCHECK -eq 1`.

So actually, perhaps your Steam check should also go here, and if Steam is missing, set `DEPSMISSING=1`. So you can copy your Steam check from `setSteamPaths` - that is, everything in `if [ ! -x "$(command -v "steam")" ] && [ "$INFLATPAK" -eq 0 ]; then` - and move it to `checkIntDeps`, inside of the `else` block. Then, instead of `exit 0`, you can set `DEPSMISSING=1`, and then we'll exit at the end of the function.

In doing this, we get a full log if an execution fails, that should list all missing dependencies.

The wiki should also be updated to note Steam as a hard dependency after this PR.

Reply:
I forgot to address your other point, the error message for Yad currently indeed does not make a whole lot of sense. I'll leave a review comment in case you want to tackle that here too, I think the fix is to replace `'$YAD'` with `Yad` (since `$YAD` is just an empty string if we get this far)